kviz/dense.py

```diff
@@ -490,6 +563,101 @@ def render(self, filename='graph'):
+        predictions.append(self.model.predict(X))
+
+        for i in range(len(X)):
+            for l in range(len(self.model.layers)):
+                layer = self.model.layers[l]
+
+                layerVals = predictions[l][i]
+                vmax = max(layerVals)
+                # multiple here to make the difference of color more obvious
+                norm = Normalize(vmin=min(layerVals), vmax=vmax)
+
+                for n in range(0, layer.input_shape[1]):
+                    act = predictions[l][i][n]
+
+                    index = unique_index(l, n)
+                    the_color_map = get_or_create_colormap_with_dict(self._graph.nodes[index]["color"], color_maps)
+
+                    if rounded:
+                        label = str(round(act, roundedN))
+                    else:
+                        label = str(act)
+
+                    if l == 0:
+                        # since this is regression, the first layer is always the input data and they do not have colors
+                        set_node_attributes(self._graph, {
+                            index: {
+                                'label': label,
+                                'fixedsize': True  # important for graphviz
+                            }})
+                    else:
+                        if act == 0:  # no color for 0
+                            color = "#ffffff"
+                        else:
+                            # minus here, so the higher value has a darker color
+                            color = str(rgb2hex(the_color_map(norm(vmax - act))))
+
+                        set_node_attributes(self._graph, {
+                            index: {
+                                'label': label,
+                                'style': 'filled',
+                                'color': color,
+                                'fixedsize': True
+                            }})
+
+                if l == len(self.model.layers) - 1:  # need to additionally show the output
+                    network_images.append(self._snap(filename))
+                    self._update_input_images_for_regression(input_images, [i], X, predictions[-1], x_color, x_marker)
+
+                    for h in range(0, layer.output_shape[1]):
+                        act = predictions[l + 1][i][h]
+
+                        if rounded:
+                            label = str(round(act, roundedN))
+                        else:
+                            label = str(act)
+
+                        index = unique_index(l + 1, h)
+
+                        set_node_attributes(self._graph, {
+                            index: {
+                                'label': label,
+                                'fixedsize': True
+                            }})
+
+                network_images.append(self._snap(filename))
+                self._update_input_images_for_regression(input_images, [i], X, predictions[-1], x_color, x_marker)
```

Comment:
what happens when the input is more than 2d? is there a nice error somewhere or does it just not show the input images?

Reply:
Updated. 
I think this should not be an error, so it does not show the input image if > 2d. The final gif will still have node activations.
For future versions maybe we can use projection (for example, tsne?) to visulize high dimension inputs.